docs/source/hyper-parameter-search.rst

```diff
@@ -420,6 +420,7 @@ This section uses :class:`~dask_ml.model_selection.HyperbandSearchCV`, but it ca
 also be applied to to :class:`~dask_ml.model_selection.IncrementalSearchCV` too.
 
 .. ipython:: python
+   :okwarning:
```

Comment:
We should open issues corresponding to the problems that we are ignoring by this. Essentially to ensure that we have it in the backlog. 


```
informative_idx, beta = dask.compute(
        informative_idx, beta, scheduler="single-threaded"
    )
```

Reply:
:100: agreed. @VibhuJawa - Do you have a reproducer? I wasn't able to get one outside of sphinx ipython directives.

Reply:
```python
import dask.distributed # sets dask.config["scheduler"] == "dask.distributed"
import dask.array as da
import dask

a = dask.compute(da.arange(10), scheduler="single-threaded")
```

`make_classification` explicitly calls (as @VibhuJawa indicates) `dask.compute` setting the scheduler to `"single-threaded"`, which then mismatches with the scheduler set up by importing `dask.distributed`.

Plausibly this is the correct fix

```diff
diff --git a/dask_ml/datasets.py b/dask_ml/datasets.py
index a561ee0d..3a5f5e99 100644
--- a/dask_ml/datasets.py
+++ b/dask_ml/datasets.py
@@ -370,9 +370,7 @@ def make_classification(
     informative_idx = rng.choice(n_features, n_informative, chunks=n_informative)
     beta = (rng.random(n_features, chunks=n_features) - 1) * scale
 
-    informative_idx, beta = dask.compute(
-        informative_idx, beta, scheduler="single-threaded"
-    )
+    informative_idx, beta = dask.compute(informative_idx, beta)
 
     z0 = X[:, informative_idx].dot(beta[informative_idx])
     y = rng.random(z0.shape, chunks=chunks[0]) < 1 / (1 + da.exp(-z0))
```

Although that backs out https://github.com/dask/dask-ml/pull/288/commits/cb5c2eec456e70a4fb39ab5d0a1c10ab28e63d86 which looks like it was a deliberate choice at the time.

Reply:
@mmccarty, **Minimal Example**

```python

from dask.distributed import Client
import dask_ml.datasets
 
client = Client()


X,y = dask_ml.datasets.make_classification_df(chunks=100)
```
```python

/datasets/vjawa/miniconda3/envs/dask-ml-dev/lib/python3.8/site-packages/dask/base.py:1282: UserWarning: Running on a single-machine scheduler when a distributed client is active might lead to unexpected results.
  warnings.warn(


```


Please note that there is a call in  `k-means` that will need to be eventually fixed too. 

https://github.com/dask/dask-ml/blob/30b3deab19755ecd45156ef9cac9fee646615100/dask_ml/cluster/k_means.py#L472-L475

Reply:
@wence- , I tried fixing it by removing the explicit cals with https://github.com/dask/dask-ml/pull/924 but that causes issues with key-names downstream [here](https://github.com/dask/dask-ml/blob/67d28b15dfff7869e9a04def203aa129a3540b27/tests/model_selection/test_split.py#L225-L226) . I suggest going with @mmccarty's fix to unblock the CI and then we can tackle this issue from the backlog.

Reply:
@VibhuJawa - Ah, of course, the warning is reproducible. Sorry, I was thinking about the key error. Opened an issue #933